website/client/src/components/inventory/items/index.vue

```diff
@@ -458,12 +465,15 @@ export default {
           }
         });
 
-        itemsArray.sort((a, b) => {
-          if (this.sortBy === 'quantity') {
-            return b.quantity - a.quantity;
-          } // AZ
-          return a.text.localeCompare(b.text);
-        });
+        if (this.sortBy === 'quantity') {
+          const quantitySnapshot = this.quantitySnapshot[groupKey] || Object.fromEntries(
```

Comment:
Given an user can have a big number of items and the rendering is already a bit slow, instead of calculating the snapshot here by iterating on the existing `itemsArray` could we do it in the `each` block at line 445 where we already iterate on the items?

Reply:
I tried moving it inside, but it doesn't seem to improve the rendering.
I've added `console.time` and `console.timeEnd` around the each + sort:
| time (ms)                | size | develop | PR   | proposal |
|-----------------|------|---------|------|----------|
| eggs            | 8    | 10.8    | 5.5  | 12.8     |
| hatchingPotions | 13   | 12.6    | 12.8 | 14.0     |
| food            | 5    | 4.3     | 3.0  | 2.7      |
| special         | 4    | 3.0     | 1.8  | 5.9      |
| quests          | 2    | 1.8     | 1.0  | 2.6      |

Adding it inside the each, requires the use of an if block,
to check if it's already cached. 
Doing the check for every element seems slower,
than running it twice, and doing the check once.
Once a snapshot is available, the time will drop even further.

Also notice that by pulling up the `sortBy` check out of the comparator,
performance has already been improved.

Reply:
thanks for checking this, let's leave this as it is then